src/irmin-pack/inode.ml

```diff
@@ -104,8 +104,12 @@ struct
     let key =
       match Conf.inode_child_order with
       | `Hash_bits ->
+          (* Bytes.unsafe_of_string usage: possibly safe TODO justify safety, or switch to
+             use the safe Bytes.of_string *)
           fun s -> Bytes.unsafe_of_string (hash_to_bin_string (Step.hash s))
       | `Seeded_hash | `Custom _ ->
+          (* Bytes.unsafe_of_string usage: possibly safe TODO justify safety, or switch to
+             use the safe Bytes.of_string *)
```

Comment:
If I am understanding the OCaml documentation correctly, these are safe because (I am assuming) we do not modify the returned buffer.

As a side thought, it could be interesting if we had a way to represent a read-only buffer in situations like this.

Reply:
Yes, I assume (hope) we don't modify the buffer either. But it wasn't clear to me that this was the case. Is it clear to you? If so, we should change the comments.

Reply:
The only way it is clear to me is based on what these are _conceptually_ representing: keys. The code could be doing other nefarious things, so I'm fine with leaving as TODO to justify this.